File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.js
/**
 * @param {number[][]} pairs
 * @return {number}
 */
var findLongestChain = function(pairs) {
    for (let i=1; i<pairs.length; i++){
        tempArr=pairs[i]
        for (var j=i-1;j>=0; j--){
            if(pairs[j][1]>tempArr[1]){
                pairs[j+1]=pairs[j]
            }
            else{
                break
            }
        }
        pairs[j+1]=tempArr
    }
    for (let k=1;k<pairs.length;k++){
        if (pairs[k-1][1]>=pairs[k][0]){
            pairs.splice(k,1)
            k--
        }
    }
    return pairs.length
};

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        for i in range(0,len(pairs)):
            min_idx=i
            for j in range(i,len(pairs)):
                if pairs[min_idx][1]>pairs[j][1]:
                    min_idx=j
            temp=pairs[i]
            pairs[i]=pairs[min_idx]
            pairs[min_idx]=temp
        k=1
        while k<len(pairs):
            if pairs[k-1][1]>=pairs[k][0]:
                pairs.pop(k)
                k-=1
            k+=1
        return len(pairs)